pkg/operator/finalizer.go

```diff
@@ -49,7 +50,16 @@ func (c *Controller) finalizeResources(o *regopapi.ImageRegistry) error {
 
 	glog.Infof("finalizing %s", objectInfo(o))
 
-	err := c.RemoveResources(o)
+	driver, err := storage.NewDriver(o.Name, o.Namespace, &o.Spec.Storage)
+	if err != nil {
+		return err
+	}
+	err = driver.RemoveStorage(o)
+	if err != nil {
+		return err
+	}
```

Comment:
you've got storage drivers returning an error when storage is not managed, which means for any user who created their own s3 buckets (and thus has unmanaged storage), they'll never be able to finalize the registry.  That's bad.

If the storage is unmanaged, I think RemoveStorage() should just return success.. it did what it's supposed to do, with respect to the storage.

If you want to maintain a distinction between removing unmanaged and removing managed storage, then you need another return value, or you need to check the error type so that we can proceed in the face of "storage is not owned by us so we couldn't remove it" errors.